legend-engine-executionPlan-execution/src/main/java/org/finos/legend/engine/plan/execution/PlanExecutor.java

```diff
@@ -80,6 +83,16 @@ private PlanExecutor(boolean isJavaCompilationAllowed, ImmutableList<StoreExecut
         this.graphFetchBatchMemoryLimit = graphFetchBatchMemoryLimit;
     }
 
+    private PlanExecutor(boolean isJavaCompilationAllowed, ImmutableList<StoreExecutor> extraExecutors, long graphFetchBatchMemoryLimit, CredentialProviderProvider credentialProviderProvider)
```

Comment:
Not specific to your change but we should think about having builder pattern for PlanExecutor.
Its in a pretty bad state